tests/driver_lis2dh12/main.c

```diff
@@ -11,170 +11,682 @@
+    .FIFO_watermark = 10,
+    .FIFO_mode = LIS2DH12_FIFO_MODE_STREAMtoFIFO,
+};
 
-/* print interrupt register */
-static void lis2dh12_int_reg_content(lis2dh12_t *dev, uint8_t pin){
+/* Memory to print current data */
+static char str_out[3][8];
 
-    assert(pin == LIS2DH12_INT1 || pin == LIS2DH12_INT2);
+/* current lis acceleration data */
+static int16_t data_lis[3];
 
-    uint8_t buffer;
-    lis2dh12_read_int_src(dev, &buffer, pin);
+/* highpass configuration */
+lis2dh12_highpass_t highpass_cfg = {
+    .Highpass_mode = LIS2DH12_HP_MODE_REFERENCE,
+    .Highpass_freq = LIS2DH12_HP_FREQ_DIV100,
+    .CLICK_enable = false,
+    .INT1_enable = false,
+    .INT2_enable = false,
+    .DATA_OUT_enable = false,
+};
 
-    printf("content SRC_Reg_%d: 0x%02x\n", pin, buffer);
-    printf("\t XL %d\n", !!(buffer & LIS2DH12_INT_SRC_XL));
-    printf("\t XH %d\n", !!(buffer & LIS2DH12_INT_SRC_XH));
-    printf("\t YL %d\n", !!(buffer & LIS2DH12_INT_SRC_YL));
-    printf("\t YH %d\n", !!(buffer & LIS2DH12_INT_SRC_YH));
-    printf("\t ZL %d\n", !!(buffer & LIS2DH12_INT_SRC_ZL));
-    printf("\t ZH %d\n", !!(buffer & LIS2DH12_INT_SRC_ZH));
-    printf("\t IA %d\n", !!(buffer & LIS2DH12_INT_SRC_IA));
-}
-#endif
+/* reference data */
+static uint8_t reference_value;
 
-int main(void)
-{
+/* shock threshold */
+static int16_t shock_thold;
 
 #ifdef MODULE_LIS2DH12_INT
-    uint8_t flags = 0;
-#endif
+/* previous values */
+static int16_t old_data_lis[3];
+static uint8_t int1_src_old;
+
+/* lis2dh12 interrupt callback function. */
+static void lis2dh12_int_cb(void* l) {
+
+    /* disable IRQ until lis_process is done */
+    gpio_irq_disable(dev.p->int1_pin);
+    gpio_irq_disable(dev.p->int2_pin);
+
+    /* reset click source */
+    lis2dh12_read_click_src(&dev, &click_src_reg);
+    DEBUG("[INT]: CLICK_SRC 0x%x\n", click_src_reg.reg);
+
+    uint8_t line = *(uint8_t*)l;
+    printf("Info: INT_line: %d\n", line);
+
+    lis2dh12_read_reference(&dev, &reference_value);
+    DEBUG("[INT]: REF: 0x%x\n", reference_value);
+
+    lis2dh12_read_int_src(&dev, &int1_src, 1);
+    DEBUG("[INT]: INT_SRC 0x%x\n", int1_src);
+    DEBUG("[INT]: INT_SRC - IA %d; ZH %d; ZL %d; YH %d; YL %d; XH %d; XL %d.\n",
+            int1_src & LIS2DH12_INT_SRC_IA,
+            int1_src & LIS2DH12_INT_SRC_ZH, int1_src & LIS2DH12_INT_SRC_ZL,
+            int1_src & LIS2DH12_INT_SRC_YH, int1_src & LIS2DH12_INT_SRC_YL,
+            int1_src & LIS2DH12_INT_SRC_XH, int1_src & LIS2DH12_INT_SRC_XL);
 
-    puts("LIS2DH12 accelerometer driver test application\n");
+    thread_wakeup(lis2dh12_process);
+}
+#endif /* MODULE_LIS2DH12_INT */
+
+void lis2dh12_test_init(void) {
 
-    puts("Initializing LIS2DH12 sensor... ");
+    /* init lis */
     if (lis2dh12_init(&dev, &lis2dh12_params[0]) == LIS2DH12_OK) {
-        puts("[OK]");
+        puts("lis2dh12 [Initialized]");
     }
     else {
-        puts("[Failed]\n");
-        return 1;
+        puts("lis2dh12 [Failed]");
```

Comment:
I was a bit confused first why the test fails on `lbc_at-v9`, then I remembered that the lis2dh12 comes in two flavors.
How about adding 

```C
if (IS_USED(MODULE_LIS2DH12_SPI)) {
    puts("using SPI mode, for I2C mode select the lis2dh12_i2c module");
} else {
    puts("using I2C mode, for SPI mode select the lis2dh12_spi module");
}
```

here